### PR TITLE
Fixes #27 : Adding User registration

### LINKED ISSUE #27: Add registration to event
https://github.com/cerebro-iiitv/cerebro-flutter/blob/46664ac1b7b8da0aa2e79d6938f6ec23332cb753/lib/screens/events.dart#L98-L102

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 46664ac1b7b8da0aa2e79d6938f6ec23332cb753 when #26 was merged. cc @yashLadha.

### PR DESCRIPTION
**Pull request by yashLadha**
- Fixes #27 

For viewing changes, you can view: [Link](https://drive.google.com/open?id=1d51tzqRR7yyhjXMIYsuJKjeyJ2qHs79P)

## Description / Changes

* Register user to event
* Option to add team name.
* Addition to team.
* Deregister from event
* Show snackbar for actions.

### CODE REVIEW

File: lib/screens/events.dart [hunk 1]
```diff
@@ -65,46 +87,350 @@ class _EventsPageState extends State<EventsPage> {
...
   }
 
+  // Prompt to show that team should be added or not
+  void showTeamPromptDialog(
+      Event event, BuildContext context, BuildContext scContext) {
+    showDialog(
+        context: context,
+        builder: (BuildContext alertContext) {
+          return AlertDialog(
+            title: Text("Prompt to add team"),
```
Comment (aashutoshrathi) [inline lib/screens/events.dart]: Make it `Confirm Team?`

File: lib/screens/events.dart [hunk 2]
```diff
@@ -65,46 +87,350 @@ class _EventsPageState extends State<EventsPage> {
...
+  // Remove user from the team
+  void removeUserFromTeam(
+      Event event, String documentID, Map<String, dynamic> data) {
+    var updateList = List();
+    for (final elem in data['members']) {
+      if (elem != curUser.id) {
+        updateList.add(elem);
+      }
+    }
+    if (updateList.length == 0) {
```
Comment (aashutoshrathi) [inline lib/screens/events.dart]: `updateList.isEmpty`

File: lib/screens/events.dart [hunk 3]
```diff
@@ -65,46 +87,350 @@ class _EventsPageState extends State<EventsPage> {
...
+        .collection(curUser.id)
+        .document('events')
+        .collection('registered')
+        .document(event.name)
+        .setData({'isRegistered': false}).whenComplete(() {
+      print('Local Registration deleted');
+      setState(() {
+        registeredEvents[event.name] = false;
+      });
+      displaySnackBar(context, "Deregistered from " + event.name);
```
Comment (aashutoshrathi) [inline lib/screens/events.dart]: Unregistered*

File: lib/screens/events.dart [hunk 4]
```diff
@@ -65,46 +87,350 @@ class _EventsPageState extends State<EventsPage> {
...
+    });
+  }
+
+  void showUnregisterPrompt(Event event, BuildContext context) {
+    showDialog(
+        context: context,
+        builder: (BuildContext dialogContext) {
+          return AlertDialog(
+            title: Text('Unrgister from Event ' + event.name),
```
Comment (aashutoshrathi) [inline lib/screens/events.dart]: Unregister :sweat_smile:

File: lib/screens/events.dart [hunk 5]
```diff
@@ -65,46 +87,350 @@ class _EventsPageState extends State<EventsPage> {
...
+              ),
+              subtitle: Text(_date(_event.date)),
+              trailing: IconButton(
+                icon: _eventBtn(_event),
+                iconSize: 30.0,
+                onPressed: () {
+                  if (registeredEvents.containsKey(_event.name)) {
+                    if (registeredEvents[_event.name])
+                      if (_event.date.difference(DateTime.now()).inDays <= 2)
+                        displaySnackBar(context, "Cannot Deregister from event " + _event.name);
```
Comment (aashutoshrathi) [inline lib/screens/events.dart]: *Unregister*

File: lib/screens/events.dart (hunk 4 above)
Comment (aashutoshrathi) [reply lib/screens/events.dart]: Also, remove `Event`

**Comment by yashLadha** [comment]
Done with the changes :+1:

**Pull request merged**